core/os/time.cpp

```diff
@@ -124,50 +124,54 @@ VARIANT_ENUM_CAST(Time::Weekday);
-		}                                                         \
-		/* Set the variables from the contents of the string. */  \
-		if (has_date) {                                           \
-			PackedInt32Array array = date.split_ints("-", false); \
-			year = array[0];                                      \
-			month = (Month)array[1];                              \
-			day = array[2];                                       \
-			/* Handle negative years. */                          \
-			if (p_datetime.find_char('-') == 0) {                 \
-				year *= -1;                                       \
-			}                                                     \
-		}                                                         \
-		if (has_time) {                                           \
-			PackedInt32Array array = time.split_ints(":", false); \
-			hour = array[0];                                      \
-			minute = array[1];                                    \
-			second = array[2];                                    \
-		}                                                         \
+#define PARSE_ISO8601_STRING(err_val)                                                            \
+	int64_t year = UNIX_EPOCH_YEAR_AD;                                                           \
+	Month month = MONTH_JANUARY;                                                                 \
+	uint8_t day = 1;                                                                             \
+	uint8_t hour = 0;                                                                            \
+	uint8_t minute = 0;                                                                          \
+	uint8_t second = 0;                                                                          \
+	{                                                                                            \
+		bool has_date = false, has_time = false;                                                 \
+		String date, time;                                                                       \
+		if (p_datetime.find_char('T') > 0) {                                                     \
+			has_date = has_time = true;                                                          \
+			PackedStringArray array = p_datetime.split("T");                                     \
+			ERR_FAIL_COND_V_MSG(array.size() < 2, err_val, "Invalid ISO-8601 date/time string"); \
+			date = array[0];                                                                     \
+			time = array[1];                                                                     \
+		} else if (p_datetime.find_char(' ') > 0) {                                              \
+			has_date = has_time = true;                                                          \
+			PackedStringArray array = p_datetime.split(" ");                                     \
+			ERR_FAIL_COND_V_MSG(array.size() < 2, err_val, "Invalid ISO-8601 date/time string"); \
```

Comment:
```suggestion
			ERR_FAIL_COND_V_MSG(array.size() < 2, err_val, "Invalid ISO 8601 date/time string"); \
```

Reply:
Oh.. I thought you mean they should be all with `-` or not. I set them to `ISO 8601`.